metrics/lib/common.bash

```diff
@@ -195,65 +172,23 @@ init_env()
 	kill_processes_before_start
 }
 
-# Clean environment, this function will try to remove all
-# stopped/running containers, it is advisable to use this function
-# in the final of each metrics test.
-clean_env()
-{
-	containers_running=$(docker ps -q)
-
-	if [ ! -z "$containers_running" ]; then
-		# First stop all containers that are running
-		# Use kill, as the containers are generally benign, and most
-		# of the time our 'stop' request ends up doing a `kill` anyway
-		sudo $DOCKER_EXE kill $containers_running
-
-		# Remove all containers
-		sudo $DOCKER_EXE rm -f $(docker ps -qa)
-	fi
-}
-
 # This function checks if there are containers or
 # shim/proxy/hypervisor processes up, if found, they are
 # killed to start test with clean environment.
 kill_processes_before_start() {
 	DOCKER_PROCS=$(${DOCKER_EXE} ps -q)
 	[[ -n "${DOCKER_PROCS}" ]] && clean_env
 
-	result=$(check_processes "$HYPERVISOR_PATH")
-	if [[ $result -ne 0 ]]; then
-		warning "Found unexpected hypervisor [${HYPERVISOR_PATH}] processes present"
-		# Sometimes we race and the process has gone by the time we list
-		# it - so make a pgrep fail non-fatal
-		pgrep -a -f "$HYPERVISOR_PATH" || true
-		sudo killall -9 "${HYPERVISOR_PATH##*/}" || true
+	check_processes
+	if [ $? -eq 0 ]; then
```

Comment:
check_processes is now a `die` case on failure I think - so we can probably never get to this kill loop. Can you confirm @GabyCT ? I probably vote we then leave check_processes as a die, and drop the kill loop here.

Reply:
@grahamwhaley perfect so for the `kill_processes_before_start`, I will leave the `check_processes` and remove the kill loop, thanks